mysql/create.sql

```diff
@@ -0,0 +1,104 @@
+CREATE DATABASE mintern;
+USE mintern;
+
+CREATE TABLE Major
+  (
+    id INT NOT NULL AUTO_INCREMENT,
+    name VARCHAR(255),
+    PRIMARY KEY (id)
+  );
+
+CREATE TABLE User
+  (
+    id INT NOT NULL AUTO_INCREMENT,
+    name VARCHAR(255),
+    email VARCHAR(255),
+    major_id INT NOT NULL,
+    is_mentor BOOLEAN,
+    PRIMARY KEY (id),
+    FOREIGN KEY (major_id) REFERENCES Major (id)
+  );
+
+CREATE TABLE Notification
+  (
+    user_id INT NOT NULL,
+    message TEXT,
+    url VARCHAR(255),
+    date DATETIME,
+    FOREIGN KEY (user_id) REFERENCES User (id)
+  );
+
+CREATE TABLE Tag
+  (
+    id INT NOT NULL AUTO_INCREMENT,
+    title VARCHAR(255),
+    color VARCHAR(255),
+    PRIMARY KEY (id)
+  );
+
+CREATE TABLE MentorExperience
+  (
+    mentor_id INT NOT NULL,
+    tag_id INT NOT NULL,
+    FOREIGN KEY (mentor_id) REFERENCES User (id),
+    FOREIGN KEY (tag_id) REFERENCES Tag (id)
+  );
+
+CREATE TABLE Question
+  (
+    id INT NOT NULL AUTO_INCREMENT,
+    title VARCHAR(255),
+    body TEXT,
+    asker_id INT NOT NULL,
+    date DATETIME,
+    PRIMARY KEY (id),
+    FOREIGN KEY (asker_id)
+    REFERENCES User (id)
+  );
+
+CREATE TABLE QuestionFollower
+  (
+    question_id INT NOT NULL,
+    follower_id INT NOT NULL,
+    FOREIGN KEY (question_id) REFERENCES Question (id),
+    FOREIGN KEY (follower_id) REFERENCES User (id)
+  );
+
+CREATE TABLE QuestionTag
+  (
+    question_id INT NOT NULL,
+    tag_id INT NOT NULL,
+    FOREIGN KEY (question_id) REFERENCES Question (id),
+    FOREIGN KEY (tag_id) REFERENCES Tag (id)
+  );
```

Comment:
i believed it wasn't necessary to have PKs in some of the tables, but since we also don't lose anything by having it i put one in every table now, thanks!

Reply:
Primary keys are always necessary. If an ID is not really needed, it could be composed of multiple attributes, including foreign keys. I'll look into them!